tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_create_workspace(self):
+        """Create Classification custom workspace."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_create_workspace")
+        inputs = {"task_type": "classification", "workspace_path": workspace_path, "otx_root": self.otx_root}
+        self.otx_builder.build_task_config(**inputs)
+        assert os.path.exists(workspace_path)
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "model.py"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data_pipeline.py"))
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_reuse_same_path(self):
+        """Raising Error of building workspace with already created path."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_create_workspace")
+        inputs = {"task_type": "classification", "workspace_path": workspace_path, "otx_root": self.otx_root}
+        with pytest.raises(FileExistsError):
+            self.otx_builder.build_task_config(**inputs)
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_normal_train_type(self):
+        """Update hparam.yaml with train_type="selfsl"."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_check_update_hparams")
+        train_type = "selfsl"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": self.otx_root,
+        }
+        self.otx_builder.build_task_config(**inputs)
+        assert os.path.exists(workspace_path)
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        expected_template_train_type = {"default_value": "SELFSUPERVISED"}
+        assert template.hyper_parameters.parameter_overrides.algo_backend.train_type == expected_template_train_type
+        model_dir = os.path.join(workspace_path, train_type)
+        assert os.path.exists(model_dir)
+        assert os.path.exists(os.path.join(model_dir, "model.py"))
+        assert os.path.exists(os.path.join(model_dir, "data_pipeline.py"))
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_abnormal_train_type(self):
+        """Raising ValueError with wrong train_type."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_abnormal_train_type")
```

Comment:
```suggestion
        workspace_path = self.tmp_dir_path / "test_builder_build_task_config_abnormal_train_type"
```

Reply:
It was modified by using `Path.joinpath`.